single_kernel_mongo/managers/upgrade.py

```diff
@@ -75,6 +76,8 @@ def _kubernetes_upgrade(self) -> None:
                 self.state.unit_upgrade_peer_data.current_revision = (
                     self.dependent.cross_app_version_checker.version  # type: ignore
                 )
+                # If the user was not existing yet, create it
```

Comment:
could you add, "we're introducing a new user that revisions predating XYZ didn't have, so we need to create it if ......"